luminoth/utils/image.py

```diff
@@ -95,20 +114,27 @@ def resize_image(image, bboxes=None, min_size=None, max_size=None):
     }
 
 
-def flip_image(image, bboxes=None, left_right=True, up_down=False):
+def flip_image(image, config, bboxes=None):
     """Flips image on its axis for data augmentation.
 
     Args:
         image: Tensor with image of shape (H, W, 3).
+        config: EasyDict
+            left_right: Boolean flag to flip the image horizontally
+                (left to right).
+            up_down: Boolean flag to flip the image vertically (upside down)
         bboxes: Optional Tensor with bounding boxes with shape
             (total_bboxes, 5).
-        left_right: Boolean flag to flip the image horizontally
-            (left to right).
-        up_down: Boolean flag to flip the image vertically (upside down)
     Returns:
         image: Flipped image with the same shape.
         bboxes: Tensor with the same shape.
     """
+    if 'left_right' not in config:
+        config.left_right = True
+    if 'up_down' not in config:
+        config.up_down = False
+    left_right = config.left_right
+    up_down = config.up_down
```

Comment:
Why change to this instead of adding `min_height` and `min_width` to kwargs in `random_patch`? It's basically replicating the kwargs functionality.

Reply:
I changed the kwargs a `config` dict because later some other data augmentation functions will probably require a much larger number of parameters, with a more complex internal structure. In particular, `random_distort` (a function that applies photometric distortion to an image) will require parameters specific to each of the possible photometric distortions applied (parameters for brightness, parameters for contrast, etc.). In that case, a config `dict` makes more sense.

Reply:
For example? What is a much larger number of parameters?

Reply:
My current `random_distortion` implementation (in a secondary branch) has the following parameter structure:
```
brightness:
    enable: Bool
    max_delta: Float
contrast:
    enable: Bool
    lower: Float
    upper: Float
hue:
    enable: Bool
    max_delta: Float
```

Reply:
Shouldn't it be then:
`random_distortion(..., brightness, contrast, hue)`
with each being dicts
?

Reply:
Aha. I now see why using `kwargs` makes more sense now. Changing it.